File: challenge #1 - Hacker language/c#/Program.cs
using System.Text;
using System.Diagnostics;

namespace Program
{
  class Program
  {
    static async Task Main(string[] args)
    {
      string inputText = await getInputText();
      const int numTasks = 1000;
      Task<IResult>[] tasks = new Task<IResult>[numTasks];

      for (int i = 0; i < numTasks; i++)
      {
        tasks[i] = Task.Run(() => leetTranslationFunction(inputText));
      }

      Task runner = Task
        .WhenAll(tasks)
        .ContinueWith(_ =>
        {
          double? totalElapsedTime = 0.0;
          double? averageElapsedTime;

          foreach (Task<IResult> task in tasks)
          {
            totalElapsedTime += task.Result.executionTimeMs;
          }

          averageElapsedTime = totalElapsedTime / numTasks;

          IResult leetTranslation = leetTranslationFunction(inputText);

          Console.WriteLine($"** result: {leetTranslation.textTranslated}");
          Console.WriteLine($"** function execution time: {leetTranslation.executionTimeMs} milliseconds");
          Console.WriteLine($"** average function execution time: {averageElapsedTime} milliseconds");

        }, TaskScheduler.Default);

      runner.Wait();
    }

    internal interface IResult
    {
      public string? textTranslated { get; set; }

      public double? executionTimeMs { get; set; }
    }

    internal class Result : IResult
    {
      public string? textTranslated { get; set; }
      public double? executionTimeMs { get; set; }
    }

    internal static string translateText(string inputText)
    {
      Dictionary<char, string> leetDictionary = new Dictionary<char, string>();

      leetDictionary.Add('A', "4");
      leetDictionary.Add('B', "|3");
      leetDictionary.Add('C', "[");
      leetDictionary.Add('D', ")");
      leetDictionary.Add('E', "3");
      leetDictionary.Add('F', "|=");
      leetDictionary.Add('G', "&");
      leetDictionary.Add('H', "#");
      leetDictionary.Add('I', "1");
      leetDictionary.Add('J', ",_|");
      leetDictionary.Add('K', ">|");
      leetDictionary.Add('L', "|_");
      leetDictionary.Add('M', "/\\/\\");
      leetDictionary.Add('N', "^/");
      leetDictionary.Add('O', "0");
      leetDictionary.Add('P', "|*");
      leetDictionary.Add('Q', "(_,)");
      leetDictionary.Add('R', "|2");
      leetDictionary.Add('S', "5");
      leetDictionary.Add('T', "7");
      leetDictionary.Add('U', "(_)");
      leetDictionary.Add('V', "\\/");
      leetDictionary.Add('W', "\\/\\/");
      leetDictionary.Add('X', "><");
      leetDictionary.Add('Y', "`/");
      leetDictionary.Add('Z', "2");
      leetDictionary.Add('1', "L");
      leetDictionary.Add('2', "R");
      leetDictionary.Add('3', "E");
      leetDictionary.Add('4', "A");
      leetDictionary.Add('5', "S");
      leetDictionary.Add('6', "b");
      leetDictionary.Add('7', "T");
      leetDictionary.Add('8', "B");
      leetDictionary.Add('9', "g");
      leetDictionary.Add('0', "()");

      StringBuilder outputText = new StringBuilder();

      // CharEnumerator enumerator = inputText.GetEnumerator();

      // while (enumerator.MoveNext())
      // {
      //   char character = enumerator.Current;

      //   if (leetDictionary.ContainsKey(char.ToUpperInvariant(character)))
      //   {
      //     outputText.Append(leetDictionary.GetValueOrDefault(char.ToUpperInvariant(character)));
      //   }
      //   else
      //   {
      //     outputText.Append(character);
      //   }
      // }

      foreach (char character in inputText)
      {
        if (leetDictionary.ContainsKey(char.ToUpperInvariant(character)))
        {
          outputText.Append(leetDictionary.GetValueOrDefault(char.ToUpperInvariant(character)));
        }
        else
        {
          outputText.Append(character);

        }
      }

      return outputText.ToString();
    }

    internal static IResult leetTranslationFunction(string inputText)
    {
      Stopwatch stopwatch = Stopwatch.StartNew();
      string textTranslated = translateText(inputText);
      stopwatch.Stop();
      double timeExecution = stopwatch.Elapsed.TotalMilliseconds;

      return new Result()
      {
        textTranslated = textTranslated,
        executionTimeMs = timeExecution,
      };
    }

    internal static async Task<string> getInputText()
    {
      return await File.ReadAllTextAsync("../text.txt");
    }

  }
}

File: challenge #1 - Hacker language/java/Program.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.TimeUnit;

public class Program {
  public static void main(String[] args) {
    Path path = Paths.get("../text.txt");
    String inputText = "";

    try {
      inputText = String.join("\n", Files.readAllLines(path));
    } catch (IOException e) {
      e.printStackTrace();
    }

    ResultTranslation resultTranslation = translateTextProfiling(inputText);

    System.out.println("result: " + resultTranslation.textTranslated);
    System.out.println("Total execution time: " + resultTranslation.duration + " milliseconds");
  }

  static class ResultTranslation {
    public String textTranslated;
    public long duration;
  }

  private static ResultTranslation translateTextProfiling(final String inputText) {
    long startTime = System.nanoTime();
    String textTranslated = translateText(inputText);
    long endTime = System.nanoTime();
    long totalTime = endTime - startTime;

    ResultTranslation resultTranslation = new ResultTranslation();
    resultTranslation.textTranslated = textTranslated;
    resultTranslation.duration = TimeUnit.NANOSECONDS.toMillis(totalTime);

    return resultTranslation;
  }

  private static String translateText(String inputText) {
    Map<Character, String> leetDictionary = new HashMap<Character, String>();

    leetDictionary.put('A', "4");
    leetDictionary.put('B', "|3");
    leetDictionary.put('C', "[");
    leetDictionary.put('D', ")");
    leetDictionary.put('E', "3");
    leetDictionary.put('F', "|=");
    leetDictionary.put('G', "&");
    leetDictionary.put('H', "#");
    leetDictionary.put('I', "1");
    leetDictionary.put('J', ",_|");
    leetDictionary.put('K', ">|");
    leetDictionary.put('L', "|_");
    leetDictionary.put('M', "/\\/\\");
    leetDictionary.put('N', "^/");
    leetDictionary.put('O', "0");
    leetDictionary.put('P', "|*");
    leetDictionary.put('Q', "(_,)");
    leetDictionary.put('R', "|2");
    leetDictionary.put('S', "5");
    leetDictionary.put('T', "7");
    leetDictionary.put('U', "(_)");
    leetDictionary.put('V', "\\/");
    leetDictionary.put('W', "\\/\\/");
    leetDictionary.put('X', "><");
    leetDictionary.put('Y', "`/");
    leetDictionary.put('Z', "2");
    leetDictionary.put('1', "L");
    leetDictionary.put('2', "R");
    leetDictionary.put('3', "E");
    leetDictionary.put('4', "A");
    leetDictionary.put('5', "S");
    leetDictionary.put('6', "b");
    leetDictionary.put('7', "T");
    leetDictionary.put('8', "B");
    leetDictionary.put('9', "g");
    leetDictionary.put('0', "()");

    String textTranslated = inputText
        .chars()
        .mapToObj(c -> leetDictionary.containsKey(Character.toUpperCase((char) c))
            ? leetDictionary.get(Character.toUpperCase((char) c))
            : (char) c)
        .collect(StringBuilder::new, StringBuilder::append, StringBuilder::append)
        .toString();

    return textTranslated;
  }
}
